src/types/Client.d.ts

```diff
@@ -26,7 +28,7 @@ export interface QueryOptions {
 
 export default class Client {
   constructor(opts?: ClientConfig)
-  query<T = object>(expr: ExprArg, options?: QueryOptions): Promise<T>
+  query<T extends object = object>(expr: ExprArg, options?: QueryOptions): Promise<T>
```

Comment:
Fauna query result is not limited to type `object` only, e.g. for the query below it returns just a plain number:
```
.query(Q.Add(2, 2)) // Promise<number>
```
Another thing is it'd be great to use the `unknown` type as a default for any generic type in Fauna. But not sure whether it's backward-compatible :thinking_face:

Reply:
removed `extends`. not sure about the `unknown` type. probably we will think about that no sooner than migrate driver code to typescript. so we can use that types during development and check how things work